src/lib.rs

```diff
@@ -1878,115 +1882,106 @@ impl GooseAttack {
     pub hatch_rate: usize,
-
-    /// Stops load test after e.g. (30s, 20m, 3h, 1h30m, etc.)
-    #[structopt(short = "t", long, required = false, default_value = "")]
+    /// Stops after (30s, 20m, 3h, 1h30m, etc)
+    #[options(short = "t", meta = "TIME")]
     pub run_time: String,
-
-    /// Doesn't track or print metrics
-    #[structopt(long)]
-    pub no_metrics: bool,
-
-    /// Doesn't track or print task metrics
-    #[structopt(long)]
-    pub no_task_metrics: bool,
-
-    /// Tracks and prints status code metrics
-    #[structopt(long)]
-    pub status_codes: bool,
-
-    /// Only prints final summary metrics in the console
-    #[structopt(long)]
-    pub only_summary: bool,
-
-    /// Resets metrics once all users have started
-    #[structopt(long)]
-    pub no_reset_metrics: bool,
-
-    /// Sets debug level (-v, -vv, -vvv, etc.)
-    #[structopt(short = "v", long, parse(from_occurrences))]
-    pub verbose: u8,
-
-    /// Sets log level (-g, -gg, -ggg, etc.)
-    #[structopt(short = "g", long, parse(from_occurrences))]
+    /// Sets log level (-g, -gg, etc)
+    #[options(short = "g", count)]
     pub log_level: u8,
-
     /// Sets log file name
-    #[structopt(long, default_value = "goose.log")]
+    #[options(default = "goose.log", help = "Sets log file name", meta = "NAME")]
     pub log_file: String,
+    #[options(
+        count,
+        short = "v",
+        // Add a blank line and then a 'Metrics:' header after this option
+        help = "Sets debug level (-v, -vv, etc)\n\nMetrics:"
+    )]
+    pub verbose: u8,
 
+    /// Only prints final summary metrics
+    #[options(no_short)]
+    pub only_summary: bool,
+    /// Doesn't reset metrics after all users have started
+    #[options(no_short)]
+    pub no_reset_metrics: bool,
+    /// Doesn't track metrics
+    #[options(no_short)]
+    pub no_metrics: bool,
+    /// Doesn't track task metrics
+    #[options(no_short)]
+    pub no_task_metrics: bool,
     /// Sets metrics log file name
-    #[structopt(short = "s", long, default_value = "")]
+    #[options(short = "m", meta = "NAME")]
     pub metrics_file: String,
-
-    /// Sets metrics log format ('csv', 'json', or 'raw')
-    #[structopt(long, default_value = "json")]
+    /// Sets metrics log format (csv, json, raw)
+    #[options(no_short, default = "json", meta = "FORMAT")]
     pub metrics_format: String,
-
     /// Sets debug log file name
-    #[structopt(short = "d", long, default_value = "")]
+    #[options(short = "d", meta = "NAME")]
     pub debug_file: String,
-
-    /// Sets debug log format ('json' or 'raw')
-    #[structopt(long, default_value = "json")]
+    /// Sets debug log format (json, raw)
+    #[options(no_short, default = "json", meta = "FORMAT")]
     pub debug_format: String,
+    // Add a blank line and then an Advanced: header after this option
+    #[options(no_short, help = "Tracks additional status code metrics\n\nAdvanced:")]
```

Comment:
That's a clever way to do it ;) - I think I would suggest to add a comment at the beginning that the order of options will correspond to the order in the output (or something like that)

Reply:
Ok, I'll add a comment as a followup